server/plugin/api.go

```diff
@@ -806,6 +806,7 @@ func (p *Plugin) searchIssues(c *UserContext, w http.ResponseWriter, r *http.Req
 	result, _, err := githubClient.Search.Issues(c.Ctx, query, &github.SearchOptions{})
 	if err != nil {
 		c.Log.WithError(err).With(logger.LogContext{"query": query}).Warnf("Failed to search for issues")
+		p.writeJSON(w, make([]*github.Issue, 0))
```

Comment:
There are a lot of other endpoints that don't return valid JSON in the error case. Should we update the webapp to not expect JSON if the status code indicates an error?

Reply:
@hanzei We were thinking of updating the plugin API response for each API where we are not returning anything in case we get any errors from the GitHub API.
For example: At [this](https://github.com/mattermost/mattermost-plugin-github/blob/40626e3a53c8d6e54026a72fbbe298a11247a0aa/server/plugin/api.go#L657) line, we can return an error something like `p.writeAPIError(w, &APIErrorResponse{Message: "Failed to search for issues", StatusCode: http.StatusInternalServerError})` and handle the error similarly at the webapp side.
What are your thoughts on this?

Reply:
If `githubClient.Search.Issues` finds nothing that matches the search, does it return an `error`?

Reply:
If we put an invalid organisation or an organisation we don't have access to, we get an error, and it gets rendered as shown in the screenshot below. 
![image](https://github.com/mattermost/mattermost-plugin-github/assets/100013900/e6d036ea-f647-442b-ba8f-ba98ed2acd82)

But it doesn't return any errors when we search for an issue with a random input term (empty or an organisation we have access to).

Reply:
We def want to avoid showing this `Unexpected end of JSON input` error in all cases

Reply:
@hanzei Should we create a new issue to update the structure of the APIs and merge this PR to resolve the issue mentioned in the description?
Also, there were two ways we were thinking of fixing the response of the API:
1. Create a wrapper and update API methods from `(c *UserContext, w http.ResponseWriter, r *http.Request)` to `(c *UserContext, r *http.Request) (any,  int)` as stated by you. We can also include a field of `error` with the data and the status code. In the wrapper, we can check for errors and data. If the error is returned, we can return `p.writeAPIError(w, &APIErrorResponse{Message: "ErrorMessage", StatusCode: statusCode})` and handle similarly on webapp and if there is no error we can return data or an empty list if data is also empty.
2. In this way, we can directly update the API to simply return `p.writeAPIError(w, &APIErrorResponse{Message: "ErrorMessage", StatusCode: statusCode})` in case of error. We can avoid making a wrapper in this case.

What are your thoughts on this? Please let me know if I missed something here.

Reply:
Either one sounds good to me. Which one is less complicated in your opinion @ayusht2810? Have you tried implemented one of them to see if it fits well?

Reply:
@mickmister We are thinking to proceed with the second method.
we can do this it in this PR or we can create a new PR and merge this so that the [issue](https://github.com/mattermost/mattermost-plugin-github/issues/557) is fixed